mesa/agent.py

```diff
@@ -36,15 +37,26 @@ class Agent:
         model (Model): A reference to the model instance.
         self.pos: Position | None = None
     """
+    _ids = itertools.count()
```

Comment:
I'm a bit confused - are agent ids unique at the model level, or at the level of each Agent type?

As a concrete example, suppose a model have agents of type A and type B. Are we allowing Type A agent with id 0 and Type B agent with id 0 at the same time?

Reply:
Just to clarify the code, and I am not hung up on the implementation details at all, `_ids` is a class attribute that is incremented using `next`. So, if this is used to get a unique_attribute within an instance as in `self.unique_id = next(self._ids)`, what happens is that `self._ids` via the MRO is resolved into the class level attribute and so returns the next number. We assign this to a instance attribute. Net result is that all subclasses of Agent will be counted and have unique numbers. 

So. 
1. With this implementation, there will be only one Agent Zero. 
2. Yes, as pointed out by @Corvince,  if you instantiate multiple models in a row, the count just keeps increasing and  is not reset with each newly instantiated model. In my view, that is actually desirable behavior (But can indeed be tricky for testing).

Reply:
Thanks for your clarification.

It is indeed very tricky. Not very certain about the differences between 2) and 3), so I might be wrong about this - from the perspective of reproducibility, is 2) better?

Basically when we re-run a model, the agent ids are unique but stay the same as previous runs (such as a uuid that can be reproduced with a random seed). Or is it too much to ask?

Reply:
I’m really fine with what everyone decides, personally

> Should it be unique to 1 Model instance?

would be enough for me.